Topics/Development_Process/Build_Requirements/CMakeLists_txt.md

```diff
@@ -0,0 +1,199 @@
+### [4. Other Notes](#other-notes)
+### [5. References](#references)
+
+---
+## Brief Introduction to CMake
+
+This article will help readers understand what `CMakeLists.txt` is, why it is used, and provide instructions on how to create a `CMakeLists.txt`. Before getting started, it is important to understand what is CMake. Note that it is presumed that the reader should have the very basic knowledge of C++, Make, and MakeFile.
+
+Unlike Make, which is a build system that drives the compiler and other build tools to build your code, CMake (or Cross-platform Make) is a "generator" of build systems that can be used for general purpose build. It is much more high-level that it can generate a low-level build script in Ninja or Make or many other build systems, and then you run it.
+
+The key feature of CMake is that it supports the cross-platform discovery of system libraries, and automatic discovery and configuration of the toolchain, which is much easier to use than Make.
+
+To install CMake, check their [document](https://cmake.org/download/) on how to download and install on Unix/Linux or Windows platform.
+
+
+## What is CMakeLists.txt?
+
+When CMake processes a project, the entry point is a source file called `CMakeLists.txt` in the top level of the source directory.
+
+`CMakeLists.txt` file is the CMake configuration file, which contains a set of directives and instructions describing the project's source files and taargets (executable, library, or both) that will be used to build the program. It determines everything for the building process, from which source files to compile, to which options to choose for the libraries and to present to the users.
+
+## Creating CMakeLists.txt
+
+### CMake Language
+`CMakeLists.txt` is written in [CMake Language](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#manual:cmake-language(7)), which consists of comments, [commands](https://cmake.org/cmake/help/latest/manual/cmake-commands.7.html#manual:cmake-commands(7)), and [variables](https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html#manual:cmake-variables(7)). 
+
+* **Comments** start with `#`
+  ```
+  # This is a comment
+  ```
+* **Commands** start with command names and a list of whitespace-separated arguments (case-sensitive).
+  ```
+  find_package(LIBIGL REQUIRED QUIET)
+  ```
+* **Variables** are case-sensitive with only alphanumeric characters and underscores. 
+
+  A number of useful variables are automatically defined by CMake (e.g `CMAKE_CURRENT_SOURCE_DIR`, `PROJECT_NAME`). 
+
+  Use *set* command to set variable names, and *${variable_name}* to reference a varible in command arguments. 
+  ```
+  set(Foo a b c)    # Value of Foo is "a;b;c"
+
+  command(${Foo})   # Foo is replaced by a;b;c 
+                    # and expands to 3 arguments
+  ```
+
+It also provides the basic **Flow Control** - Conditional statements, Looping constructs, and Procedure definitions. Here are the examples:
```

Comment:
Should probably by "control flow" rather than "Flow Control"? "Flow control" is a completely different concept.

Also, not sure why "Conditional statements, Looping constructs, and Procedure definitions" has capitalization

Reply:
I see. This term is quite confusing and is mostly used in _data communication_ (like TCP flow control). I didn't realize the issue when I wrote this, and was simply following the term used on this [website](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Writing%20CMakeLists%20Files.html).
You are correct. It should be replaced by any of the "control flow", "flow of control", "flow-control elements" or "control flow constructs" that are more commonly used in this situation. I'll fix that using the "control flow".

For the second part, I don't know if that's important but I'll make them lower case too.